app/models/middleware_server_performance.rb

```diff
@@ -16,7 +16,11 @@ def self.build_results_for_report_middleware(options)
     start_time, end_time, interval = parse_time_interval(options)
     results = []
     MiddlewareServer.find_each do |ms|
-      raw_stats = fetch_raw_stats(ms, start_time, end_time, interval)
+      begin
+        raw_stats = fetch_raw_stats(ms, start_time, end_time, interval)
```

Comment:
I think there are two aspects here:
- the connection error, which we can for sure do as a specific exception (inside the Hawkular gem namespace)
- the catching of exceptions in the loop, which may or may not be to catch broader exceptions than the connection one only. It probably makes sense to "just abort" when the issue is really not inside the Hawkular code or responsibility.